app/routes/auth.py

```diff
@@ -0,0 +1,133 @@
+import time
+import uuid
+from typing import Annotated
+
+import requests
+from fastapi import APIRouter, Depends, HTTPException, Query, Response, status
+from fastapi.security import OAuth2PasswordRequestForm
+from sqlalchemy.orm import Session
+
+from app import crud, schemas
+from app.auth import oauth2_scheme
+from app.config import settings
+from app.db import get_db
+from app.models import Session as SessionModel
+
+auth_router = APIRouter(prefix="/auth", tags=["Auth"])
+session_router = APIRouter(prefix="/session", tags=["Auth"])
+
+
+def create_token(user_id: str):
+    return f"{user_id}__U{str(uuid.uuid4())}"
+
+
+def get_current_session(
+    token: Annotated[str, Depends(oauth2_scheme)], db: Session = Depends(get_db)
+) -> SessionModel:
+    """Get the current user session, if authenticated.
+
+    This function is used as a dependency in endpoints that require
+    authentication. It raises an HTTPException if the user is not
+    authenticated.
+
+    :param token: the authentication token
+    :param db: the database session
+    :raises HTTPException: if the user is not authenticated
+    :return: the current user session
+    """
+    if token and "__U" in token:
+        session = crud.get_session_by_token(db, token=token)
+        if session:
+            return crud.update_session_last_used_field(db, session=session)
+    raise HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Invalid authentication credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+
+
+@auth_router.post("", tags=["Auth"])
```

Comment:
```suggestion
@auth_router.post("")
```
no need to repeat the `tags` info now that it is defined at the top :)